src/common/transformations/src/transformations/sdpa_to_paged_attention/total_sequence_length_pattern.cpp

```diff
@@ -23,21 +23,62 @@ ov::pass::TotalSequenceLengthPattern::TotalSequenceLengthPattern(
     auto kv_current = pattern::any_input();
     auto kv_concat = pattern::wrap_type<v0::Concat>({kv_gather, kv_current});
     auto kv_shape = pattern::wrap_type<v3::ShapeOf>({kv_concat});
-    auto seq = pattern::wrap_type<v8::Gather>({kv_shape, pattern::any_input(), pattern::any_input()});
+    auto gather_idx_label = pattern::wrap_type<v0::Constant>();
+    auto seq = pattern::wrap_type<v8::Gather>({kv_shape, gather_idx_label, pattern::any_input()});
 
     ov::matcher_pass_callback callback = [=](ov::pass::pattern::Matcher& m) {
         // TODO: Check that seq has axis that really takes sequence len but not any other dimension --
         //  use symbolic infra or look at the constant input
-        auto gather = m.get_match_root();
-        auto target_type = gather->get_output_element_type(0);
-        std::shared_ptr<Node> replacement = max_context_len;
-        if (replacement->get_output_element_type(0) != target_type) {
-            replacement = std::make_shared<v0::Convert>(replacement, target_type);
+        const auto& pattern_map = m.get_pattern_value_map();
+
+        auto concat = std::dynamic_pointer_cast<v0::Concat>(pattern_map.at(kv_concat).get_node_shared_ptr());
+        auto gather = pattern_map.at(seq).get_node_shared_ptr();
+        auto gather_idx =
+            std::dynamic_pointer_cast<v0::Constant>(pattern_map.at(gather_idx_label).get_node_shared_ptr());
+        auto gather_idx_data = gather_idx->cast_vector<int64_t>();
```

Comment:
please use `gather_idx-> `  after line 42
theoretically `gather_idx` might be nullptr here

Reply:
Sorry, didn't quite get you here. Why would `gather_idx ` be nullptr? I check for it @40. And why not to use it after line 42?

Reply:
std::dynamic_pointer_cast<v0::Constant> might return `nullptr` if it can't cast it

Reply:
but it might be at line 38, you use it here ` gather_idx->cast_vector<int64_t>();`

Reply:
Oh, you're right! Good catch. I'll fix it!

Reply:
Fixed.